### PR TITLE
Feature/#344 Rework PositionHandler for all scenarios

### PR DESCRIPTION
**Pull request by ggrg**
## Coverage summary
Statements   : 50.38% ( 1117/2217 )
Branches     : 35.46% ( 161/454 )
Functions    : 35.6% ( 152/427 )
Lines        : 51.08% ( 1114/2181 )

### 1..362
tests 362
pass  362
**ok**

### Code coverage report for src/handlers/positions/handler.js
100% Statements 66/66
100% Branches 30/30
100% Functions 4/4
100% Lines 66/66

### CODE REVIEW

File: src/handlers/positions/handler.js [hunk 1]
```diff
@@ -59,21 +74,39 @@ const positions = async (error, messages) => {
...
     const payload = message.value.content.payload
-    if (message.value.metadata.event.type === POSITION && message.value.metadata.event.action === PREPARE) {
+    if (message.value.metadata.event.type === TransferEventType.POSITION && message.value.metadata.event.action === TransferEventAction.PREPARE) {
+      // Consumed Prepare message for Payer
+      consumer = Kafka.Consumer.getConsumer(Utility.transformAccountToTopicName(message.value.from, TransferEventType.POSITION, TransferEventAction.PREPARE))
       await Projection.updateTransferState(payload, TransferState.RESERVED)
-    } else if (message.value.metadata.event.type === POSITION && message.value.metadata.event.action === COMMIT) {
+    } else if (message.value.metadata.event.type === TransferEventType.POSITION && message.value.metadata.event.action === TransferEventAction.COMMIT) {
+      // Consumed Commit message for Payee
+      consumer = Kafka.Consumer.getConsumer(Utility.transformAccountToTopicName(message.value.from, TransferEventType.POSITION, TransferEventAction.COMMIT))
```
Comment (rmothilal) [inline src/handlers/positions/handler.js]: TransferEventAction.COMMIT needs to be TransferEventType.FULFIL

**Comment by rmothilal** [comment]
Your integration test failed

File: config/default.json [hunk 2]
```diff
@@ -181,7 +232,7 @@
               "retry.backoff.ms": 100,
               "message.send.max.retries": 2,
               "socket.keepalive.enable": true,
-              "queue.buffering.max.messages": 10,
+              "queue.buffering.max.messages": 6000,
```
Comment (rmothilal) [inline config/default.json]: maybe change this back to 10

File: config/default.json [hunk 3]
```diff
@@ -203,7 +254,7 @@
...
```
Comment (rmothilal) [inline config/default.json]: maybe change this back to 10

File: config/default.json [hunk 4]
```diff
@@ -225,7 +276,7 @@
...
```
Comment (rmothilal) [inline config/default.json]: maybe change this back to 10

File: config/default.json [hunk 5]
```diff
@@ -247,7 +298,7 @@
               "retry.backoff.ms": 100,
               "message.send.max.retries": 2,
               "socket.keepalive.enable": true,
-              "queue.buffering.max.messages": 10,
+              "queue.buffering.max.messages": 200,
```
Comment (rmothilal) [inline config/default.json]: maybe change this back to 10

File: config/default.json [hunk 6]
```diff
@@ -271,7 +322,51 @@
...
```
Comment (rmothilal) [inline config/default.json]: maybe change this back to 10

File: config/default.json [hunk 7]
```diff
@@ -271,7 +322,51 @@
...
+            "rdkafkaConf": {
+              "debug": "all",
+              "metadata.broker.list": "localhost:9092",
+              "client.id": "default-client",
+              "event_cb": true,
+              "compression.codec": "none",
+              "retry.backoff.ms": 100,
+              "message.send.max.retries": 2,
+              "socket.keepalive.enable": true,
+              "queue.buffering.max.messages": 200,
```
Comment (rmothilal) [inline config/default.json]: maybe change this back to 10

File: src/handlers/positions/handler.js [hunk 8]
```diff
@@ -82,23 +115,19 @@ const positions = async (error, messages) => {
...
- * @function CreatePositionHandler
- *
- * @async
- * @description Registers the handler for each participant topic created. Gets Kafka config from default.json
- *
- * Calls createHandler to register the handler against the Stream Processing API
- * @returns {boolean} - Returns a boolean: true if successful, or throws and error if failed
- */
-const createPositionHandler = async (participantName) => {
+const createPositionHandlerForPrepare = async (participantName) => {
```
Comment (rmothilal) [inline src/handlers/positions/handler.js]: change name to createPositionHandlers

File: src/handlers/positions/handler.js [hunk 9]
```diff
@@ -118,10 +147,12 @@ const registerPositionHandlers = async () => {
     const participantList = await DAO.retrieveAllParticipants()
     if (participantList.length !== 0) {
       for (let name of participantList) {
-        await createPositionHandler(name)
+        await createPositionHandlerForPrepare(name)
```
Comment (rmothilal) [inline src/handlers/positions/handler.js]: change name to createPositionHandlers

File: config/default.json (hunk 2 above)
Comment (ggrg) [reply config/default.json]: done

File: config/default.json (hunk 3 above)
Comment (ggrg) [reply config/default.json]: done

File: config/default.json (hunk 4 above)
Comment (ggrg) [reply config/default.json]: done

File: config/default.json (hunk 5 above)
Comment (ggrg) [reply config/default.json]: done

File: config/default.json (hunk 6 above)
Comment (ggrg) [reply config/default.json]: done

File: config/default.json (hunk 7 above)
Comment (ggrg) [reply config/default.json]: done

File: src/handlers/positions/handler.js (hunk 8 above)
Comment (ggrg) [reply src/handlers/positions/handler.js]: done

File: src/handlers/positions/handler.js (hunk 9 above)
Comment (ggrg) [reply src/handlers/positions/handler.js]: done

**Comment by ggrg** [comment]
@rmothilal, I fixed the issues you requested. I also run integration tests locally once again and it run successfully!

**Comment by ggrg** [comment]
Is there any chance default.json should be copied separately to CircleCI?

**Pull request merged**